packages/extension/src/background/handlers/Extension.ts

```diff
@@ -246,25 +246,30 @@ export default class Extension {
     return true;
   }
 
-  private derive (parentAddress: string, suri: string, metadata: KeyringPair$Meta): KeyringPair {
+  private derive (parentAddress: string, suri: string, password: string, metadata: KeyringPair$Meta): KeyringPair {
     const parentPair = keyring.getPair(parentAddress);
-    return parentPair.derive(suri, metadata);
-  }
-
-  private derivationValidate ({ parentAddress, suri }: RequestDeriveValidate): ResponseDeriveValidate {
     try {
-      const childPair = this.derive(parentAddress, suri, {});
-      return {
-        address: childPair.address,
-        suri
-      };
+      parentPair.decodePkcs8(password);
+    } catch (e) {
+      throw new Error(`"${password}" is not a valid password`);
```

Comment:
```suggestion
      throw new Error(`invalid password`);
```
Imagine entering password ******** and getting error: hunder2 is not a valid password :)

Reply:
Thought that it is only internal error. But right, it might leak to frontend somehow and it won't be nice 😛